components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/certificatemgt/CertificateManagerImpl.java

```diff
@@ -333,6 +333,26 @@ public boolean isConfigured() {
         return true;
     }
 
+    @Override
+    public CertificateMetadataDTO getCertificate(String alias, int tenantId) {
+
+        List<CertificateMetadataDTO> certificateMetadataList = null;
+        CertificateMetadataDTO certificate = null;
+
+        try {
+            if (log.isDebugEnabled()) {
+                log.debug("Retrieving certificate metadata for alias: " + alias + " and tenant:-> " + tenantId);
+            }
+            certificateMetadataList = certificateMgtDAO.getCertificates(alias, "", tenantId);
+            if (certificateMetadataList != null && certificateMetadataList.size() == 1) {
+                certificate = certificateMetadataList.get(0);
+            }
+        } catch (CertificateManagementException e) {
+            log.error("Error when retrieving certificate metadata for alias '" + alias + "'", e);
```

Comment:
is it sufficient to log the error and continue the flow? don't we need to handle it?

Reply:
Yeah it gives a null certificate which is handled later